skxray/cdi.py

```diff
@@ -119,9 +142,284 @@ def convolution(array1, array2):
     `this issue on github <https://github.com/Nikea/scikit-xray/issues/258>`_
     for details.
     """
-    fft_norm = lambda x:  np.fft.fftshift(np.fft.fftn(x)) / np.sqrt(np.size(x))
-    fft_1 = fft_norm(array1)
-    fft_2 = fft_norm(array2)
-    return np.abs(np.fft.ifftshift(np.fft.ifftn(fft_1*fft_2)) *
-                  np.sqrt(np.size(array2)))
+    fft_1 = _fft_helper(array1) / np.sqrt(np.size(array1))
+    fft_2 = _fft_helper(array2) / np.sqrt(np.size(array2))
+    return np.abs(_fft_helper(fft_1*fft_2, option='ifft')) * np.sqrt(np.size(array2))
+
+
+def pi_modulus(array_in, diff_array,
```

Comment:
As a general rule, drastically changing the code-path of a function based on a string switch is an anti-pattern.